src/workerd/api/tests/compat-flags-test.js

```diff
@@ -0,0 +1,15 @@
+import {
+  ok,
+} from 'node:assert';
+
+import { usingCompatFlag } from 'cloudflare:workers';
+
+export const compatFlags = {
+  test() {
+    ok(usingCompatFlag('nodejs_compat_v2'));
```

Comment:
if I use this as `usingCompatFlag('no_nodejs_compat_v2')` will `false` result mean that `node_compat_v2` is not active or that the user didn't configure workerd with `no_nodejs_compat_v2` (which are the exact opposites).

Could we tweak the API such that this ambiguity doesn't exist?

Reply:
have you considered to using imported boolean constants instead of `usingCompatFlag` and string arg here? For example:

```
import { NODEJS_COMPAT_V2 } from 'cloudflare:workers';

...
if (NODEJS_COMPAT_V2) {
  console.log('yay! you are using nodejs_compat_v2 mode!!');
}
```

Reply:
> if I use this as usingCompatFlag('no_nodejs_compat_v2') will false result mean that node_compat_v2 is not active or that the user didn't configure workerd with no_nodejs_compat_v2 (which are the exact opposites).

If you specify the `nodejs_compat_v2` flag... then `usingCompatFlag('no_nodejs_compat_v2')` would return `false` and `usingCompatFlag('nodejs_compat_v2')` would return true. Inversely, if you specify `no_nodejs_compat_v2` then `usingCompatFlag('nodejs_compat_v2')` would be false and `usingCompatFlag('no_nodejs_compat_v2')` would be true.

> have you considered to using imported boolean constants instead of usingCompatFlag and string arg here?

Yeah, that's basically what we do for the internal modules. I didn't go that route here to avoid adding too many new symbols exported on the module. I can go either way with it but I tend to prefer this approach. If someone wants const values they can easily define those themselves. Then again, constants would perform better since those would be initialized only once and there's no need for the caching.

Reply:
Ok, updated to replace the method with an object full of constants...

```js
import { compatFlags } from 'cloudflare:workers';

console.log(compatFlags['nodejs_compat_v2']);
```

The `compatFlags` is an object like...

```
{
  formdata_parser_supports_files: true,
  formdata_parser_converts_files_to_strings: false,
  fetch_refuses_unknown_protocols: true,
  fetch_treats_unknown_protocols_as_http: false,
  html_rewriter_treats_esi_include_as_void_tag: false,
  durable_object_fetch_requires_full_url: true,
  durable_object_fetch_allows_relative_url: false,
  streams_byob_reader_detaches_buffer: true,
  streams_byob_reader_does_not_detach_buffer: false,
  streams_enable_constructors: true,
  streams_disable_constructors: false,
  workers_api_getters_setters_on_prototype: true,
  workers_api_getters_setters_on_instance: false,
  minimal_subrequests: true,
  no_minimal_subrequests: false,
  no_cots_on_external_fetch: true,
  cots_on_external_fetch: false,
  url_standard: true,
  url_original: false,
  global_navigator: true,
  no_global_navigator: false,
  capture_async_api_throws: true,
  do_not_capture_async_api_throws: false,
  r2_public_beta_bindings: true,
  r2_internal_beta_bindings: false,
  durable_object_alarms: false,
  dont_substitute_null_on_type_error: true,
  substitute_null_on_type_error: false,
  transformstream_enable_standard_constructor: true,
  transformstream_disable_standard_constructor: false,
  r2_list_honor_include: true,
  export_commonjs_default: true,
  export_commonjs_namespace: false,
  durable_object_rename: false,
  web_socket_compression: false,
  no_web_socket_compression: true,
  nodejs_compat: false,
  no_nodejs_compat: true,
  tcp_sockets_support: false,
  response_redirect_url_standard: false,
  response_redirect_url_original: true,
  experimental: false,
  durable_object_get_existing: false,
  http_headers_getsetcookie: false,
  no_http_headers_getsetcookie: true,
  dynamic_dispatch_tunnel_exceptions: false,
  dynamic_dispatch_treat_exceptions_as_500: true,
  service_binding_extra_handlers: false,
  no_cf_botmanagement_default: false,
  cf_botmanagement_default: true,
  urlsearchparams_delete_has_value_arg: false,
  no_urlsearchparams_delete_has_value_arg: true,
  strict_compression_checks: false,
  no_strict_compression_checks: true,
  brotli_content_encoding: false,
  no_brotli_content_encoding: true,
  strict_crypto_checks: false,
  no_strict_crypto_checks: true,
  rtti_api: false,
  webgpu: false,
  crypto_preserve_public_exponent: false,
  no_crypto_preserve_public_exponent: true,
  vectorize_query_metadata_optional: false,
  vectorize_query_original: true,
  unsafe_module: false,
  js_rpc: false,
  no_global_importscripts: false,
  global_importscripts: true,
  nodejs_als: false,
  no_nodejs_als: true,
  queues_json_messages: false,
  no_queues_json_messages: true,
  python_workers: false,
  fetcher_no_get_put_delete: false,
  fetcher_has_get_put_delete: true,
  unwrap_custom_thenables: false,
  no_unwrap_custom_thenables: true,
  rpc: false,
  no_rpc: true,
  internal_stream_byob_return_view: false,
  internal_stream_byob_return_undefined: true,
  blob_standard_mime_type: false,
  blob_legacy_mime_type: true,
  fetch_standard_url: false,
  fetch_legacy_url: true,
  nodejs_compat_v2: true,
  no_nodejs_compat_v2: false
}
```